daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/unparsers/UState.scala

```diff
@@ -344,7 +342,21 @@ abstract class UState(
       // Finished means you won't add data to the end of it any more.
       // It does NOT prevent information like the absoluteBitPos to
       // propagate.
-      dos.setFinished(this)
+      //
+      // Need this clone or we could possibly end up in an infinite loop inside of
+      // DirectOrBuffereDataOutputStream.cleanUp()
+      val finfo = this match {
+        case m: UStateMain => m.cloneForSuspension(dos)
+        case _ => this
```

Comment:
I wonder if this case is even possible? If we are adding a new buffered DOS below, I think that means that this must be a `UStateMain`--we can't split a DOS that is associated with an already cloned UState since that means the UState is from a suspension which must have already been split.  I think `this` *must* be a `UStateMain` and that it *must* be cloned. I think we can just change `this` to something like `Assert.invariantFailed("state must be a UStateMain when splitting for bit order change")`